packages/transit-vehicle-overlay/src/VehicleGeometry/index.js

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { FeatureGroup } from "react-leaflet";
+import { leafletPathType } from "@opentripplanner/core-utils/lib/types";
+import { vehicleType } from "../types";
+import * as utils from "./utils";
+import { setColor } from "../utils";
+
+/**
+ * vehicle geometry presentational component that creates a map overlay for the line
+ * geometry showing the travel pattern of a vehicle
+ */
+function VehicleGeometry(props) {
+  const { trackedVehicle } = props;
+  const { pattern } = props;
+  const { lowlight, color } = props;
+  let { highlight } = props;
+  if (color) {
+    highlight = setColor(color, highlight);
+  }
+
+  let retVal = <FeatureGroup />;
+  if (trackedVehicle && pattern && pattern.data) {
+    const pt = utils.findPointOnLine(trackedVehicle, pattern.data);
+    const geom = utils.splitGeometry(pattern.data, pt, pattern.id);
+    const segments = utils.makeSplitLine(geom, highlight, lowlight);
```

Comment:
The calculations from the above lines probably can be cached somehow, but that might be something for a later effort.